aii-core/src/main/perl/Shellfe.pm

```diff
@@ -308,6 +326,28 @@ sub _initialize
     if ($self->option(INCLUDE)) {
         unshift(@INC, split(/:+/, $self->option(INCLUDE)));
     }
+
+    # Initialization of some options whose default values are based on other options
+    if ( !$self->option(NBPDIR_GRUB) ) {
+        if ( $self->option(NBPDIR_PXELINUX) ) {
+            my $efi_dir = dirname($self->option(NBPDIR_PXELINUX)) . '/grub-efi';
+            $self->{CONFIG}->set(NBPDIR_GRUB, $efi_dir);
+        } else {
+            $self->{CONFIG}->set(NBPDIR_GRUB, '/osinstall/nbp/grub-efi');
+        }
+    }
+    if ( !$self->option(GRUB_EFI_KERNEL_ROOT) ) {
+        my $kernel_root = '';
+        if ( $self->option(NBPDIR_PXELINUX) ) {
+            $kernel_root = dirname($self->option(NBPDIR_PXELINUX));
+        } elsif ( $self->option(NBPDIR_GRUB) ) {
+            $kernel_root = dirname($self->option(NBPDIR_GRUB));
+        }
+        $kernel_root =~ s/^\/\w+//;
```

Comment:
This is explained in the issue. Grub looks for it with a path starting at the TFTPD root, not the directory where it is installed (which is PXELINUX behaviour). Thus the path present in the prefix for the installation kernel/initrd must generally be prefixed by something that cannot be derived unambiguously from another option.... Thus the need for this one.